test/meta_store_test.rb

```diff
@@ -190,6 +190,19 @@ def self.call(request); request.path_info.reverse end
         mock.verify
       end
 
+      it 'purges meta store entry when the entry does not contain the digest header' do
+        cache_key = store_simple_entry
+        meta_entry = @store.read(cache_key)
+        meta_entry.select { |e| e.is_a?(Array) }.flatten.each { |h| h.is_a?(Hash) && h.delete('x-content-digest') }
```

Comment:
Minor: [Enumerable#grep](https://rubyapi.org/3.3/o/enumerable#method-i-grep) uses `===` to compare, and can take a class to do "`#select` + `#is_a?`".

```suggestion
        meta_entry.grep(Array).flatten.each { |h| h.is_a?(Hash) && h.delete('x-content-digest') }
```

The power of threequals.

Reply:
https://github.com/rack/rack-cache/pull/22/commits/70282c9eedd182a74df78ea198b24ffe88ab9b4f